Xamarin.PropertyEditing.Mac/Controls/ControlExtensions.cs

```diff
@@ -37,6 +37,9 @@ public static Task<ITypeInfo> RequestAt (this TypeRequestedEventArgs self, IHost
 			popover.SetAppearance (hostResources.GetVibrantAppearance (source.EffectiveAppearance));
 
 			tcs.Task.ContinueWith (t => {
+				// Focus back to the button that popped us up
+				source.Superview.Window.MakeFirstResponder (source);
```

Comment:
Is there any cancellation support? This can break, as nothing guarantees the view or window is still displayed/valid by the time the continuation is called.

Reply:
@Therzok Can you clarify what you mean by "cancelation support"?

Reply:
In example, tcs.Task.ContinueWith means that the continuation callback will run after task.Task is finished.

In case we have something like:
```csharp
source.Dispose();
await tcs.Task; // this will run the continuation callback
```

this will crash on executing source.Superview in the continuation above, simply because we've disposed the view in the meantime. A CancellationTokenSource should be bound to the view and cancelled when the view is disposed.

Reply:
There really shouldn't ever be a scenario in which that happens. Worst case I'd say let's just safeguard the continuation, we don't need to set up a whole cancellation setup for this.

Reply:
@ermau are you talking about just ensuring `source` isn't null at this point or some other belt and braces?

Reply:
The superview and the window. `source` can't be null, it's a closed over local that's never set to null.

Reply:
Done.